### PR TITLE
Implement layered configuration loading with OrthoConfig, loader, and tests

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- Adds a full layered configuration system for podbot using OrthoConfig derives, enabling the precedence: defaults < file < environment < CLI.
- Introduces a dedicated loader bridging OrthoConfig with the crate's error model, updates AppConfig to derive OrthoConfig, and wires configuration loading into the application startup flow.
- Adds an OrthoConfig error bridge, environment variable mapping, and export of loading helpers.
- Expands tests (unit, integration, and BDD) to validate layered precedence, and updates documentation to reflect discovery and precedence.
- Includes an exec plan doc for the OrthoConfig derive, plus roadmap and user guide updates.

### Core configuration
- Add layered config support via OrthoConfig derives on AppConfig and nested configs, with a PostMergeHook placeholder for future normalisation.
- Introduce a loader module (src/config/loader.rs) to load and merge defaults, config file, environment variables, and CLI overrides in the correct order.
- Update src/config/mod.rs to export `load_config` and the new loader (env_var_names and load_config).
- Add an OrthoConfig bridge in error handling (src/error.rs) with a new variant wrapping ortho_config errors.

### CLI and application wiring
- Update src/main.rs to load configuration via `load_config` and pass AppConfig into command handlers.
- Minor updates to src/config/cli.rs to reflect layered precedence and discovery comments (no breaking CLI API changes).

### Tests
- Expand unit tests in src/config/tests.rs to cover layered precedence using MergeComposer (defaults, file, env, CLI).
- Extend BDD tests (tests/bdd_config.rs, helpers, and features) to cover layered precedence scenarios (file overrides defaults, env overrides file, CLI overrides all).
- Integration tests added: tests/load_config_integration.rs to validate load_config API end-to-end.
- Tests have been reorganised and expanded into modular files (layer precedence tests, helpers, types, validation).

### Documentation
- New exec plan documenting OrthoConfig derive for layered precedence (docs/execplans/1-3-6-ortho-config-derive.md).
- Update docs/podbot-roadmap.md to reflect completed items for configuration module.
- Update docs/users-guide.md to describe configuration discovery order and discovery paths.

### Artefacts and notes
- The changes introduce a new loader that bridges OrthoConfig with the crate’s error model, plus a richer test matrix including integration and BDD tests.
- Public API exposure remains compatible: `load_config()` and `env_var_names()` are exported from `podbot::config`; `ConfigError::OrthoConfig` is added for ortho_config-related failures.

## Why this approach
- Provides robust, testable, and scalable config management across multiple sources with clear precedence.
- Keeps configuration logic isolated in a dedicated loader while leveraging OrthoConfig for declarative, derive-based handling.
- Enables thorough testing of precedence rules via MergeComposer and BDD scenarios.

## Validation and testing
- Unit tests validate defaults, file overrides, env overrides, CLI overrides, nested merges, and empty-layer edge cases.
- Integration/BDD tests cover layered precedence paths and ensure correct precedence semantics across defaults, file, env, and CLI.
- Documentation updates provide discovery paths and precedence rules.

## How to test locally
- Build and test the project:
  - cargo test
  - make check-fmt && make lint
- Run configuration-related tests and features:
  - cargo test --tests
  - Ensure BDD scenarios pass (tests/features/configuration.feature) and that new layered precedence tests pass.
- Validate runtime loading:
  - Run podbot with various combinations of config.toml, PODBOT_CONFIG_PATH, PODBOT_ENGINE_SOCKET, and PODBOT_IMAGE; verify that CLI args override environment and file-based settings, and environment overrides file merges as expected.

## Notes
- This change introduces a new OrthoConfig-based loading path and a loader that bridges with the crate’s error model. It is designed to be non-breaking to the public CLI surface while centralising and clarifying configuration precedence.
- Future work could enable more granular PostMergeHook normalisation and expand discovery to additional config sources if needed.

📎 **Task**: https://www.terragonlabs.com/task/bc0e24a9-f40e-4d4c-8673-94cb2b394fa2

### CODE REVIEW

File: src/lib.rs.tmp [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// Temp check
+use ortho_config::OrthoConfig;
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/lib.rs.tmp]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

~~~
Please address the comments from this code review:

## Overall Comments
- The new loader uses `composer.push_defaults(json!({}))` while the tests build defaults from `AppConfig::default()`; consider switching the loader to serialise `AppConfig::default()` as the defaults layer to keep behaviour consistent and avoid subtle divergence if `Default` and `serde(default)` ever differ.
- In `loader.rs`, invalid `PODBOT_*` environment variable values (e.g., non-bool for `PODBOT_SANDBOX_PRIVILEGED`, non-u64 for GitHub IDs) are silently ignored; if that is not intentional, consider surfacing these as `ConfigError` variants so misconfigurations are visible rather than being dropped.
- The new `src/lib.rs.tmp` file looks like a temporary artefact and should probably be removed from the repository.

## Individual Comments

### Comment 1
<location> `src/config/types.rs:180-189` </location>
<code_context>
+4. Add OrthoConfig derive to `AppConfig` with discovery:
+
+   ```rust
+   #[derive(Debug, Clone, Default, Deserialize, Serialize, OrthoConfig)]
+   #[ortho_config(
+       prefix = "PODBOT",
+       post_merge_hook,
+       discovery(
+           app_name = "podbot",
+           env_var = "PODBOT_CONFIG_PATH",
+           config_file_name = "config.toml",
+           dotfile_name = ".podbot.toml",
+           config_cli_long = "config",
+           config_cli_short = 'c',
+           config_cli_visible = true,
+       )
+   )]
+   pub struct AppConfig { ... }
</code_context>

<issue_to_address>
**suggestion:** Avoid duplicating OrthoConfig discovery/merging logic between the derive and the custom loader to reduce drift.

`AppConfig` now derives `OrthoConfig` with full `discovery(...)` settings, but `load_config` in `loader.rs` still performs its own discovery, env handling, and CLI override logic. That duplication risks the two paths drifting (e.g., when adding fields or changing env/discovery rules). Consider routing `load_config` through the derive-generated APIs or another shared abstraction so config discovery and env mapping are defined in one place.
</issue_to_address>

### Comment 2
<location> `src/config/loader.rs:62-71` </location>
<code_context>
+    // Layer 2: Configuration file.
+    // Use the CLI-provided path, or discover via XDG paths.
+    let config_path: Option<PathBuf> = cli.config.as_ref().map(|p| p.as_std_path().to_owned());
+    let discovered_path = config_path.or_else(|| {
+        // Discover config files using ortho_config's ConfigDiscovery builder.
+        let discovery = ConfigDiscovery::builder("podbot")
+            .env_var("PODBOT_CONFIG_PATH")
+            .config_file_name("config.toml")
+            .dotfile_name(".podbot.toml")
+            .build();
+        discovery.candidates().into_iter().find(|p| p.exists())
+    });
+
+    if let Some(path) = discovered_path.filter(|p| p.exists()) {
+        load_config_file(&path, &mut composer)?;
+    }
</code_context>

<issue_to_address>
**nitpick:** The `exists()` check is performed twice when resolving the config file path.

The `or_else` branch already narrows candidates to existing paths via `find(|p| p.exists())`, so the later `discovered_path.filter(|p| p.exists())` is redundant. You can simplify this to `if let Some(path) = discovered_path { ... }` to keep the control flow clearer without changing behavior.
</issue_to_address>

### Comment 3
<location> `src/config/tests.rs:79-80` </location>
<code_context>
     }
 }

+/// Helper: Creates a `MergeComposer` with defaults layer already pushed.
+fn create_composer_with_defaults() -> MergeComposer {
+    let mut composer = MergeComposer::new();
+    let defaults = ortho_config::serde_json::to_value(AppConfig::default())
</code_context>

<issue_to_address>
**suggestion (testing):** Consider mirroring the production defaults behavior in these layer-precedence tests

Right now the tests use `create_composer_with_defaults`, which serializes `AppConfig::default()` into the `defaults` layer, while `load_config` uses `composer.push_defaults(json!({}))` and relies on OrthoConfig/serde defaults. If these ever differ, the tests could still pass while the real loading path diverges. Please add at least one test that uses the same defaults mechanism as `load_config` (an empty JSON defaults layer) and checks that the resulting `AppConfig` equals `AppConfig::default()`.
</issue_to_address>

### Comment 4
<location> `tests/bdd_config_helpers.rs:359-361` </location>
<code_context>
+// Layer precedence step definitions
+
+/// Merges the current file layer with a new value (combining fields).
+fn merge_file_layer(
+    config_state: &ConfigState,
+    new_value: ortho_config::serde_json::Value,
+) -> ortho_config::serde_json::Value {
+    if let Some(existing) = config_state.file_layer.get() {
</code_context>

<issue_to_address>
**suggestion:** Consider extracting a shared merge helper and/or handling nested objects to make BDD layer merges more robust

`merge_file_layer` and `merge_env_layer` both do a shallow merge with nearly identical logic. A shared helper like `merge_layer(existing: &Slot<Value>, new_value: Value) -> Value` would reduce duplication and make future changes safer.

Also, with the current shallow merge, any later step that sets a nested object (e.g. `sandbox`, `github`) will replace the whole object instead of merging fields. If you expect multi-step scenarios on nested structures, a recursive, key-by-key merge (mirroring how `OrthoConfig` merges nested config) would avoid surprising overwrites and make the BDD behavior closer to real config resolution.

Suggested implementation:

```rust
fn merge_json_values(
    existing: &ortho_config::serde_json::Value,
    new_value: &ortho_config::serde_json::Value,
) -> ortho_config::serde_json::Value {
    use ortho_config::serde_json::Value;

    match (existing, new_value) {
        (Value::Object(existing_obj), Value::Object(new_obj)) => {
            let mut merged = existing_obj.clone();

            for (key, new_child) in new_obj {
                if let Some(existing_child) = merged.get(key) {
                    // Recursively merge nested objects; for non-objects the new value wins.
                    merged.insert(
                        key.clone(),
                        merge_json_values(existing_child, new_child),
                    );
                } else {
                    merged.insert(key.clone(), new_child.clone());
                }
            }

            Value::Object(merged)
        }
        // For non-object values, the new value completely overwrites the existing one.
        _ => new_value.clone(),
    }
}

fn merge_layer(
    existing: Option<&ortho_config::serde_json::Value>,
    new_value: ortho_config::serde_json::Value,
) -> ortho_config::serde_json::Value {
    if let Some(existing_value) = existing {
        merge_json_values(existing_value, &new_value)
    } else {
        new_value
    }
}

/// Merges the current file layer with a new value (combining fields).
fn merge_file_layer(
    config_state: &ConfigState,
    new_value: ortho_config::serde_json::Value,
) -> ortho_config::serde_json::Value {
    let existing = config_state.file_layer.get();
    merge_layer(existing.as_ref(), new_value)
}

```

To fully apply the refactoring and DRY up the logic, you should also:

1. Update `merge_env_layer` (and any other similar helpers) to use the shared `merge_layer` function, for example:

```rust
fn merge_env_layer(
    config_state: &ConfigState,
    new_value: ortho_config::serde_json::Value,
) -> ortho_config::serde_json::Value {
    let existing = config_state.env_layer.get();
    merge_layer(existing.as_ref(), new_value)
}
```

2. If there are tests that depend on the previous shallow merge semantics (especially around nested objects), adjust them to reflect the new recursive merge behavior where nested config (e.g. `sandbox`, `github`) is merged field-by-field instead of being entirely replaced by later steps.
</issue_to_address>

### Comment 5
<location> `tests/bdd_config.rs:117-126` </location>
<code_context>
     let _ = config_state;
 }
+
+// Layer precedence scenarios
+
+#[scenario(
+    path = "tests/features/configuration.feature",
+    name = "File layer overrides defaults"
+)]
+fn file_layer_overrides_defaults(config_state: ConfigState) {
+    let _ = config_state;
+}
+
+#[scenario(
+    path = "tests/features/configuration.feature",
+    name = "Environment layer overrides file layer"
+)]
+fn env_layer_overrides_file(config_state: ConfigState) {
+    let _ = config_state;
+}
+
+#[scenario(
+    path = "tests/features/configuration.feature",
+    name = "CLI layer overrides all other layers"
+)]
+fn cli_layer_overrides_all(config_state: ConfigState) {
+    let _ = config_state;
+}
+
+#[scenario(
+    path = "tests/features/configuration.feature",
+    name = "Lower layer values are preserved when not overridden"
+)]
+fn lower_layer_values_preserved(config_state: ConfigState) {
+    let _ = config_state;
+}
</code_context>

<issue_to_address>
**suggestion (testing):** Add a BDD scenario that exercises nested config precedence (e.g. sandbox/github) to mirror unit tests

Unit tests in `src/config/tests.rs` already verify nested merging (e.g. `sandbox.privileged`, `github.app_id`/`installation_id`). To mirror that end-to-end, please add a Gherkin scenario plus bindings that cover a nested object—for example, a file layer setting `sandbox.privileged = true` with an env layer overriding it to `false` while preserving `mount_dev_fuse`. That way the Cucumber tests exercise the same nested precedence semantics as the unit tests.

Suggested implementation:

```rust
#[scenario(
    path = "tests/features/configuration.feature",
    name = "File layer overrides defaults"
)]
fn file_layer_overrides_defaults(config_state: ConfigState) {
    let _ = config_state;
}

#[scenario(
    path = "tests/features/configuration.feature",
    name = "Nested configuration layer precedence"
)]
fn nested_configuration_layer_precedence(config_state: ConfigState) {
    let _ = config_state;
}

```

To fully implement the requested behavior, you also need to:

1. Add a matching Gherkin scenario to `tests/features/configuration.feature`, e.g.:

   ```gherkin
   Scenario: Nested configuration layer precedence
     Given a config file with:
       | key                    | value |
       | sandbox.privileged     | true  |
       | sandbox.mount_dev_fuse | true  |
     And environment variables:
       | key                | value |
       | SANDBOX_PRIVILEGED | false |
     When the configuration is loaded
     Then the merged configuration should have:
       | key                    | value |
       | sandbox.privileged     | false |
       | sandbox.mount_dev_fuse | true  |
   ```

2. Ensure your existing step definitions (or new ones if necessary) can:
   - Set nested config keys from the file layer (e.g. `sandbox.privileged`, `sandbox.mount_dev_fuse`).
   - Set corresponding environment overrides (e.g. `SANDBOX_PRIVILEGED`).
   - Assert on the final merged config values, including nested fields, consistent with the unit tests in `src/config/tests.rs` (e.g. how `sandbox.privileged`, `github.app_id`, and `github.installation_id` are merged).
</issue_to_address>

### Comment 6
<location> `src/config/loader.rs:37` </location>
<code_context>
+use crate::config::AppConfig;
+use crate::error::{ConfigError, PodbotError, Result};
+
+/// Load configuration with full layer precedence.
+pub fn load_config() -> Result<AppConfig> {
+    AppConfig::load().map_err(|e| {
</code_context>

<issue_to_address>
**issue (complexity):** Consider refactoring the config loader to rely more on the `OrthoConfig` derive (and, if needed, a data‑driven env spec table) instead of custom discovery, merging, and per‑section env collectors.

You can simplify this module substantially by delegating more to the `OrthoConfig` derive and/or making the env mapping data‑driven instead of bespoke per section.

### 1. Let `OrthoConfig` handle discovery + env

You’re reconstructing logic that `AppConfig` already knows via `#[derive(OrthoConfig)]` with `discovery(...)` and `prefix = "PODBOT"`.

Instead of:

```rust
let mut composer = MergeComposer::new();
composer.push_defaults(json!({}));

// manual discovery
let config_path: Option<PathBuf> = cli.config.as_ref().map(|p| p.as_std_path().to_owned());
let discovered_path = config_path.or_else(|| {
    let discovery = ConfigDiscovery::builder("podbot")
        .env_var("PODBOT_CONFIG_PATH")
        .config_file_name("config.toml")
        .dotfile_name(".podbot.toml")
        .build();
    discovery.candidates().into_iter().find(|p| p.exists())
});
if let Some(path) = discovered_path.filter(|p| p.exists()) {
    load_config_file(&path, &mut composer)?;
}

// manual env + CLI JSON
let env_values = collect_env_vars();
if !env_values.is_null() { composer.push_environment(env_values); }
let cli_overrides = build_cli_overrides(cli);
if !cli_overrides.is_null() { composer.push_cli(cli_overrides); }

let config = AppConfig::merge_from_layers(composer.layers())?;
```

you can push almost all of this into a single call and only deal with the very small CLI override surface:

```rust
pub fn load_config(cli: &Cli) -> Result<AppConfig> {
    // Let OrthoConfig handle defaults + discovery + env
    let mut config = AppConfig::load().map_err(ConfigError::OrthoConfig)?;

    // Apply the few global CLI overrides
    if let Some(ref socket) = cli.engine_socket {
        config.engine_socket = socket.clone();
    }
    if let Some(ref image) = cli.image {
        config.image = image.clone();
    }

    Ok(config)
}
```

This keeps your precedence semantics (defaults < file < env < CLI) but removes the need for `MergeComposer`, `ConfigDiscovery`, and custom env JSON in this module.

If you need to pass the CLI‑provided `--config` path into `AppConfig::load`, you can expose an explicit API on `AppConfig` instead of re‑implementing discovery here:

```rust
impl AppConfig {
    pub fn load_with_path(path: Option<&std::path::Path>) -> Result<Self, ortho_config::Error> {
        // inside here, you can reuse the derive-generated discovery
        // but allow an explicit "override path" when provided
    }
}

pub fn load_config(cli: &Cli) -> Result<AppConfig> {
    let path = cli.config.as_ref().map(|p| p.as_std_path());
    let mut config = AppConfig::load_with_path(path).map_err(ConfigError::OrthoConfig)?;

    if let Some(ref socket) = cli.engine_socket {
        config.engine_socket = socket.clone();
    }
    if let Some(ref image) = cli.image {
        config.image = image.clone();
    }

    Ok(config)
}
```

### 2. If you must keep custom env mapping, make it data‑driven

If there are requirements that force you to stay with `MergeComposer` + JSON env mapping, you can still collapse the many `collect_*_env_vars` helpers into a single table‑driven function, which will be easier to maintain and less tightly coupled to struct renames.

Instead of six bespoke functions like:

```rust
fn collect_github_env_vars(map: &mut Map<String, Value>) { /* ... */ }
fn collect_sandbox_env_vars(map: &mut Map<String, Value>) { /* ... */ }
// etc.
```

you can use a small description table:

```rust
struct EnvSpec<'a> {
    env: &'a str,
    path: &'a [&'a str],       // JSON path: ["github", "app_id"]
    parse: fn(String) -> Option<serde_json::Value>,
}

fn parse_bool(s: String) -> Option<Value> {
    s.parse::<bool>().ok().map(Value::Bool)
}

fn parse_u64(s: String) -> Option<Value> {
    s.parse::<u64>().ok().map(|n| Value::Number(n.into()))
}

fn parse_string(s: String) -> Option<Value> {
    Some(Value::String(s))
}

fn collect_env_vars() -> Value {
    use serde_json::{Map, Value};

    const SPECS: &[EnvSpec] = &[
        EnvSpec { env: "PODBOT_ENGINE_SOCKET", path: &["engine_socket"], parse: parse_string },
        EnvSpec { env: "PODBOT_IMAGE",         path: &["image"],         parse: parse_string },
        EnvSpec { env: "PODBOT_GITHUB_APP_ID", path: &["github", "app_id"], parse: parse_u64 },
        EnvSpec { env: "PODBOT_GITHUB_INSTALLATION_ID", path: &["github", "installation_id"], parse: parse_u64 },
        EnvSpec { env: "PODBOT_GITHUB_PRIVATE_KEY_PATH", path: &["github", "private_key_path"], parse: parse_string },
        EnvSpec { env: "PODBOT_SANDBOX_PRIVILEGED", path: &["sandbox", "privileged"], parse: parse_bool },
        EnvSpec { env: "PODBOT_SANDBOX_MOUNT_DEV_FUSE", path: &["sandbox", "mount_dev_fuse"], parse: parse_bool },
        EnvSpec { env: "PODBOT_AGENT_KIND", path: &["agent", "kind"], parse: parse_string },
        EnvSpec { env: "PODBOT_AGENT_MODE", path: &["agent", "mode"], parse: parse_string },
        EnvSpec { env: "PODBOT_WORKSPACE_BASE_DIR", path: &["workspace", "base_dir"], parse: parse_string },
        EnvSpec { env: "PODBOT_CREDS_COPY_CLAUDE", path: &["creds", "copy_claude"], parse: parse_bool },
        EnvSpec { env: "PODBOT_CREDS_COPY_CODEX", path: &["creds", "copy_codex"], parse: parse_bool },
    ];

    let mut root = Map::new();

    for spec in SPECS {
        if let Ok(raw) = std::env::var(spec.env) {
            if let Some(value) = (spec.parse)(raw) {
                insert_path(&mut root, spec.path, value);
            }
        }
    }

    if root.is_empty() { Value::Null } else { Value::Object(root) }
}

fn insert_path(root: &mut Map<String, Value>, path: &[&str], value: Value) {
    let mut current = root;

    for key in &path[..path.len() - 1] {
        current = match current.entry((*key).to_owned()).or_insert_with(|| Value::Object(Map::new())) {
            Value::Object(map) => map,
            _ => return, // path conflict, ignore or handle as needed
        };
    }

    current.insert(path[path.len() - 1].to_owned(), value);
}
```

This preserves the current behavior (same env names, same JSON shape) but removes the tight coupling between multiple ad‑hoc functions and the config layout, and makes adding/changing variables a single‑line change in the `SPECS` table.
</issue_to_address>

### Comment 7
<location> `src/config/loader.rs:53` </location>
<code_context>
+/// - Malformed configuration files
+/// - Invalid environment variable values
+/// - Missing required fields after merge
+pub fn load_config(cli: &Cli) -> Result<AppConfig> {
+    let mut composer = MergeComposer::new();
+
</code_context>

<issue_to_address>
**issue (review_instructions):** Add both unit tests and behavioural (BDD) tests that exercise the new layered configuration loading behaviour implemented in `load_config`.

Add unit tests (e.g. in `src/config/tests.rs`) that verify each precedence layer for `load_config` – defaults, config file, environment variables, and CLI overrides – and ensure they compose correctly into `AppConfig`.

Also add behavioural/BDD scenarios (e.g. in `tests/features/configuration.feature` with corresponding step definitions) that cover user-visible configuration behaviour, such as:
- `--config` and `PODBOT_CONFIG_PATH` affecting which file is loaded,
- environment variables overriding file values,
- CLI flags overriding both env and file.

Do not merge this change without these tests, as the instructions require both behavioural and unit tests for this new configuration feature.

<details>
<summary>Review instructions:</summary>

**Path patterns:** `**/*`

**Instructions:**
Rules:
- For any new feature or change to an existing feature, both behavioural *and* unit tests are required.
- Bug fixes *must* be demonstrated by a test.
- Keep code DRY, but readable. Use refactoring approaches best suited for the language in question. Context managers and generators for Python; RAII and macros for Rust.

</details>
</issue_to_address>

### Comment 8
<location> `docs/users-guide.md:98` </location>
<code_context>
+
+1. Path specified via `--config` CLI argument
+2. Path specified via `PODBOT_CONFIG_PATH` environment variable
+3. `$XDG_CONFIG_HOME/podbot/config.toml` (usually
+   `~/.config/podbot/config.toml`)
+4. `~/.podbot.toml` (dotfile in home directory)
</code_context>

<issue_to_address>
**suggestion (review_instructions):** The acronym "XDG" in `$XDG_CONFIG_HOME` is used without explanation on first use in this guide section.

Consider briefly defining XDG at its first mention, for example: "the XDG configuration directory (`$XDG_CONFIG_HOME`, usually `~/.config`)" so readers unfamiliar with XDG understand the reference.

<details>
<summary>Review instructions:</summary>

**Path patterns:** `**/*.md`

**Instructions:**
Define uncommon acronyms on first use.

</details>
</issue_to_address>
~~~

File: tests/load_config_integration.rs [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+/// - No concurrent access to these environment variables is occurring
+fn clear_podbot_env() {
+    for var in PODBOT_ENV_VARS {
+        // SAFETY: Tests are run serially via `#[serial]` attribute,
+        // preventing concurrent access to environment variables.
+        unsafe {
+            std::env::remove_var(var);
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/load_config_integration.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

~~~
❌ Not Resolved (1/11)
Comments 1 & 6 (core issue): OrthoConfig derive delegation

Status: Not resolved

The loader remains at 332 lines and still manually implements:

    Discovery via ConfigDiscovery::builder (lines 194-199)
    Layer composition via MergeComposer (line 180, 187, 206, etc.)
    Custom merging logic

This duplicates the functionality already provided by #[derive(OrthoConfig)] with discovery(...) settings on AppConfig. The architecture remains tightly coupled to the manual implementation rather than delegating to the derive-generated APIs.

Note: The data-driven env spec table (Comment 6, second suggestion) was implemented, but the main suggestion to delegate discovery and merging to OrthoConfig remains unaddressed.
~~~

File: src/config/tests.rs [hunk 3]
```diff
@@ -74,6 +76,20 @@ fn github_config_complete() -> GitHubConfig {
...
+    let defaults = ortho_config::serde_json::to_value(AppConfig::default())
+        .expect("serialization should succeed");
+    composer.push_defaults(defaults);
+    composer
+}
+
+/// Helper: Merges layers from a composer into `AppConfig`.
+fn merge_config(composer: MergeComposer) -> AppConfig {
+    AppConfig::merge_from_layers(composer.layers()).expect("merge should succeed")
+}
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/config/tests.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: tests/load_config_integration.rs [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+    #[case] invalid_value: &str,
+    #[case] expected_type: &str,
+) {
+    let _guard = clear_podbot_env();
+
+    // SAFETY: Mutex guard ensures exclusive access to environment variables.
+    unsafe {
+        std::env::set_var(env_var, invalid_value);
+    }
+
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/load_config_integration.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix:

[src/config/tests/types_tests.rs](https://github.com/leynos/podbot/pull/19/files/4d1bbe46282b7930a2aca1713777bb2733fb8d98#diff-282b378289da3e6f25614ba476cdf32b86a2103cdd35b17abac42fe87be35c4e)

Comment on lines +94 to +124

~~~rust
fn app_config_toml_sets_engine_socket_and_image() {
    let toml = r#"
        engine_socket = "unix:///run/podman/podman.sock"
        image = "ghcr.io/example/sandbox:latest"

        [github]
        app_id = 12345
        installation_id = 67890

        [sandbox]
        privileged = true
        mount_dev_fuse = false

        [agent]
        kind = "codex"
        mode = "podbot"

        [workspace]
        base_dir = "/home/user/work"
    "#;

    let config: AppConfig = toml::from_str(toml).expect("TOML parsing should succeed");
    assert_eq!(
        config.engine_socket.as_deref(),
        Some("unix:///run/podman/podman.sock")
    );
    assert_eq!(
        config.image.as_deref(),
        Some("ghcr.io/example/sandbox:latest")
    );
}
~~~

❌ New issue: [Code Duplication](https://codescene.io/projects/75835/delta?repo-id=1423864&review-id=19&biomarker=Code+Duplication&filename=src%2Fconfig%2Ftests%2Ftypes_tests.rs&method=)
The module contains 5 functions with similar structure: app_config_toml_sets_agent_config,app_config_toml_sets_engine_socket_and_image,app_config_toml_sets_github_ids,app_config_toml_sets_sandbox_flags and 1 more functions
